src/sqlitestorage.cpp

```diff
@@ -1169,6 +1171,8 @@ int SqliteStorage::Private::loadIncidences(sqlite3_stmt *stmt1,
             //                  << incidence->dtStart() << endDateTime
             //                  << "in calendar";
             count += 1;
+        } else {
+            delete incidence;
```

Comment:
Is the above break; case leaking memory now since selectComponents() return value is owned by the caller? Pondering should it still just return Incidence::Ptr since incidences are going to end up as such and there are these cases where it's ignored.

Reply:
You're right. I've reverted these changes at the moment. Ideally it's in preparation of SqliteStorage not assuming that loaded incidences will be used in the same thread. But these changes can be done at that moment.